### ISSUE TITLE
iOS - Setting digits as text property value of ActionBar item crashes the app

### ISSUE THREAD

**Issue by vhristov5555** [NativeScript/nativescript-angular#911]
Code snippet:
```
<ActionBar title="444" class="action-bar">
</ActionBar>
```

Output:
```
2017-07-20 11:51:23.184 foo[840:256049] -[__NSCFNumber stringByTrimmingCharactersInSet:]: unrecognized selector sent to instance 0xb0000000000014d2
2017-07-20 11:51:28.201 foo[840:256049] *** JavaScript call stack:
(
	0   setTitle@[native code]
	1   update@file:///app/tns_modules/tns-core-modules/ui/action-bar/action-bar.js:106:23
	2   ActionBarComponent@file:///app/tns_modules/nativescript-angular/directives/action-bar.js:57:35
	3   createClass@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:10902:34
	4   createDirectiveInstance@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:10730:48
	5   createViewNodes@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:12093:72
	6   callViewAction@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:12537:28
	7   execComponentViewsAction@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:12446:27
	8   createViewNodes@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:12120:29
	9   createRootView@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:11998:20
	10  create@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:9919:60
	11  createComponent@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:10121:68
	12  activateOnGoForward@file:///app/tns_modules/nativescript-angular/router/page-router-outlet.js:153:70
	13  activateWith@file:///app/tns_modules/nativescript-angular/router/page-router-outlet.js:139:37
	14  placeComponentIntoOutlet@file:///app/tns_modules/@angular/router/./bundles/router.umd.js:4526:28
	15  activateRoutes@file:///app/tns_modules/@angular/router/./bundles/router.umd.js:4507:50
	16  @file:///app/tns_modules/@angular/router/./bundles/router.umd.js:4443:72
	17  forEach@:1:11
	18  activateChildRoutes@file:///app/tns_modules/@angular/router/./bundles/router.umd.js:4443:36
	19  activate@file:///app/tns_modules/@angular/router/./bundles/router.umd.js:4417:33
	20  @file:///app/tns_modules/@angular/router/./bundles/router.umd.js:4034:30
	21  @file:///app/tns_modules/rxjs/Observable.js:110:25
	22  __tryOrSetError@file:///app/tns_modules/rxjs/Subscriber.js:247:20
	23  next@file:///app/tns_modules/rxjs/Subscriber.js:187:42
	24  _next@file:///app/tns_modules/rxjs/Subscriber.js:125:30
	25  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	26  _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	27  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	28  notifyNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:143:34
	29  _next@file:///app/tns_modules/rxjs/InnerSubscriber.js:23:31
	30  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	31  _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	32  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	33  _subscribe@file:///app/tns_modules/rxjs/observable/ScalarObservable.js:49:28
	34  _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	35  subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	36  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	37  _innerSub@file:///app/tns_modules/rxjs/operator/mergeMap.js:130:55
	38  _tryNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:127:23
	39  _next@file:///app/tns_modules/rxjs/operator/mergeMap.js:110:26
	40  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	41  notifyNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:143:34
	42  _next@file:///app/tns_modules/rxjs/InnerSubscriber.js:23:31
	43  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	44  _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	45  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	46  _complete@file:///app/tns_modules/rxjs/operator/reduce.js:119:34
	47  complete@file:///app/tns_modules/rxjs/Subscriber.js:114:27
	48  _complete@file:///app/tns_modules/rxjs/operator/mergeMap.js:135:38
	49  complete@file:///app/tns_modules/rxjs/Subscriber.js:114:27
	50  _subscribe@file:///app/tns_modules/rxjs/observable/ArrayObservable.js:116:32
	51  _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	52  subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	53  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	54  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	55  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	56  _innerSub@file:///app/tns_modules/rxjs/operator/mergeMap.js:130:55
	57  _tryNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:127:23
	58  _next@file:///app/tns_modules/rxjs/operator/mergeMap.js:110:26
	59  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	60  notifyNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:143:34
	61  _next@file:///app/tns_modules/rxjs/InnerSubscriber.js:23:31
	62  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	63  _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	64  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	65  notifyNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:143:34
	66  _next@file:///app/tns_modules/rxjs/InnerSubscriber.js:23:31
	67  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	68  notifyComplete@file:///app/tns_modules/rxjs/operator/every.js:53:30
	69  _complete@file:///app/tns_modules/rxjs/operator/every.js:70:28
	70  complete@file:///app/tns_modules/rxjs/Subscriber.js:114:27
	71  _complete@file:///app/tns_modules/rxjs/operator/mergeMap.js:135:38
	72  complete@file:///app/tns_modules/rxjs/Subscriber.js:114:27
	73  _subscribe@file:///app/tns_modules/rxjs/observable/ArrayObservable.js:116:32
	74  _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	75  subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	76  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	77  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	78  _innerSub@file:///app/tns_modules/rxjs/operator/mergeMap.js:130:55
	79  _tryNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:127:23
	80  _next@file:///app/tns_modules/rxjs/operator/mergeMap.js:110:26
	81  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	82  notifyComplete@file:///app/tns_modules/rxjs/operator/every.js:53:30
	83  _complete@file:///app/tns_modules/rxjs/operator/every.js:70:28
	84  complete@file:///app/tns_modules/rxjs/Subscriber.js:114:27
	85  _complete@file:///app/tns_modules/rxjs/operator/mergeMap.js:135:38
	86  complete@file:///app/tns_modules/rxjs/Subscriber.js:114:27
	87  _subscribe@file:///app/tns_modules/rxjs/observable/ArrayObservable.js:116:32
	88  _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	89  subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	90  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	91  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	92  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	93  subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	94  _innerSub@file:///app/tns_modules/rxjs/operator/mergeMap.js:130:55
	95  _tryNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:127:23
	96  _next@file:///app/tns_modules/rxjs/operator/mergeMap.js:110:26
	97  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	98  _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	99  next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	100 notifyNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:143:34
	101 _next@file:///app/tns_modules/rxjs/InnerSubscriber.js:23:31
	102 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	103 _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	104 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	105 _subscribe@file:///app/tns_modules/rxjs/observable/ScalarObservable.js:49:28
	106 _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	107 subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	108 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	109 _innerSub@file:///app/tns_modules/rxjs/operator/mergeMap.js:130:55
	110 _tryNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:127:23
	111 _next@file:///app/tns_modules/rxjs/operator/mergeMap.js:110:26
	112 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	113 notifyNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:143:34
	114 _next@file:///app/tns_modules/rxjs/InnerSubscriber.js:23:31
	115 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	116 _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	117 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	118 _subscribe@file:///app/tns_modules/rxjs/observable/ScalarObservable.js:49:28
	119 _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	120 subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	121 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	122 _innerSub@file:///app/tns_modules/rxjs/operator/mergeMap.js:130:55
	123 _tryNext@file:///app/tns_modules/rxjs/operator/mergeMap.js:127:23
	124 _next@file:///app/tns_modules/rxjs/operator/mergeMap.js:110:26
	125 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	126 notifyNext@file:///app/tns_modules/rxjs/OuterSubscriber.js:19:30
	127 _next@file:///app/tns_modules/rxjs/InnerSubscriber.js:23:31
	128 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	129 _next@file:///app/tns_modules/rxjs/Subscriber.js:125:30
	130 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	131 _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	132 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	133 _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	134 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	135 _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	136 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	137 _complete@file:///app/tns_modules/rxjs/operator/last.js:110:29
	138 complete@file:///app/tns_modules/rxjs/Subscriber.js:114:27
	139 _complete@file:///app/tns_modules/rxjs/operator/mergeAll.js:94:38
	140 complete@file:///app/tns_modules/rxjs/Subscriber.js:114:27
	141 _subscribe@file:///app/tns_modules/rxjs/observable/ScalarObservable.js:51:36
	142 _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	143 subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	144 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	145 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	146 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	147 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	148 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	149 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	150 error@file:///app/tns_modules/rxjs/operator/catch.js:111:59
	151 _error@file:///app/tns_modules/rxjs/Subscriber.js:128:31
	152 error@file:///app/tns_modules/rxjs/Subscriber.js:102:24
	153 _error@file:///app/tns_modules/rxjs/Subscriber.js:128:31
	154 error@file:///app/tns_modules/rxjs/Subscriber.js:102:24
	155 error@file:///app/tns_modules/rxjs/operator/catch.js:107:44
	156 _error@file:///app/tns_modules/rxjs/Subscriber.js:128:31
	157 error@file:///app/tns_modules/rxjs/Subscriber.js:102:24
	158 notifyError@file:///app/tns_modules/rxjs/OuterSubscriber.js:22:31
	159 _error@file:///app/tns_modules/rxjs/InnerSubscriber.js:26:32
	160 error@file:///app/tns_modules/rxjs/Subscriber.js:102:24
	161 _error@file:///app/tns_modules/rxjs/Subscriber.js:128:31
	162 error@file:///app/tns_modules/rxjs/Subscriber.js:102:24
	163 error@file:///app/tns_modules/rxjs/operator/catch.js:107:44
	164 _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	165 subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	166 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	167 _next@file:///app/tns_modules/rxjs/operator/mergeAll.js:85:59
	168 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	169 _next@file:///app/tns_modules/rxjs/operator/map.js:83:30
	170 next@file:///app/tns_modules/rxjs/Subscriber.js:89:23
	171 _subscribe@file:///app/tns_modules/rxjs/observable/ArrayObservable.js:114:32
	172 _trySubscribe@file:///app/tns_modules/rxjs/Observable.js:57:35
	173 subscribe@file:///app/tns_modules/rxjs/Observable.js:45:40
	174 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	175 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	176 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	177 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	178 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	179 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	180 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	181 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	182 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	183 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	184 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	185 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	186 subscribe@file:///app/tns_modules/rxjs/Observable.js:42:26
	187 @file:///app/tns_modules/rxjs/Observable.js:89:43
	188 ZoneAwarePromise@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:776:37
	189 forEach@file:///app/tns_modules/rxjs/Observable.js:85:31
	190 @file:///app/tns_modules/@angular/router/./bundles/router.umd.js:4015:25
	191 ZoneAwarePromise@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:776:37
	192 runNavigate@file:///app/tns_modules/@angular/router/./bundles/router.umd.js:3946:27
	193 onInvoke@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:4156:43
	194 run@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:125:49
	195 @file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:760:60
	196 onInvokeTask@file:///app/tns_modules/@angular/core/./bundles/core.umd.js:4147:47
	197 runTask@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:165:57
	198 drainMicroTaskQueue@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:593:42
	199 promiseReactionJob@:1:11
)
2017-07-20 11:51:28.211 foo[840:256049] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFNumber stringByTrimmingCharactersInSet:]: unrecognized selector sent to instance 0xb0000000000014d2'
*** First throw call stack:
(0x183b942d8 0x1953b80e4 0x183b9b3a4 0x183b98154 0x183a9accc 0x188679b14 0x100d98044 0x100d9711c 0x100d96c34 0x1004cd4fc 0x100b03c50 0x100a80580 0x100c34ea8 0x100c81cf0 0x100afa540 0x100b069e4 0x100b0a228 0x100b0a5c0 0x100b0a1c4 0x100b0a1c4 
```

**Comment by NickIliev** [comment]
@vhristov5555 are you still reproducing this issue!?
I have tested via [this app ](https://github.com/NickIliev/NativeScript-Issues-2017/tree/master/nativescript-angular/issue_911)and it works as expected on both iOS and Android.

Note: Try using the latest released modules 3.1.0

**Comment by dtopuzov** [comment]
I can confirm this is fixed in @next.
